### ISSUE TITLE
SC1090 can't be satisfied -- it wants shellcheck directives (which are shell comments)

### ISSUE THREAD

**Issue by ianfixes** [hadolint/hadolint#343]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Minimal example

Consider this `mini.Dockerfile`

```Dockerfile
FROM alpine:3
SHELL ["/bin/bash", "-c"]
ENV NVM_DIR=/usr/local/nvm
RUN source $NVM_DIR/nvm.sh
```

Linting fails:

```console
$ docker run --rm -i hadolint/hadolint:latest /bin/hadolint - < mini.Dockerfile 
/dev/stdin:4 SC1090 Can't follow non-constant source. Use a directive to specify location.
```

### What does `shellcheck` say?

According to https://github.com/koalaman/shellcheck/wiki/SC1090 the fix is to put a shell comment "directive" to assist:

```shell
# shellcheck source=src/lib.sh
. "$(find_install_dir)/lib.sh"
```

> If you don't care that ShellCheck is unable to account for the file, specify `# shellcheck source=/dev/null`.

### But that doesn't work...

`mini2.Dockerfile`

```Dockerfile
FROM alpine:3
SHELL ["/bin/bash", "-c"]
ENV NVM_DIR=/usr/local/nvm
RUN # shellcheck source=/dev/null
RUN source $NVM_DIR/nvm.sh
```

```console
$ docker run --rm -i hadolint/hadolint:latest /bin/hadolint - < mini2.Dockerfile 
/dev/stdin:4 SC1072 Expected a command. Fix any mentioned problems and try again.
/dev/stdin:5 SC1090 Can't follow non-constant source. Use a directive to specify location.
```

The old "Now you have 2 problems"...

`mini3.Dockerfile`

```Dockerfile
FROM alpine:3
SHELL ["/bin/bash", "-c"]
ENV NVM_DIR=/usr/local/nvm
# shellcheck source=/dev/null
RUN source $NVM_DIR/nvm.sh
```

```console
$ docker run --rm -i hadolint/hadolint:latest /bin/hadolint - < mini3.Dockerfile 
/dev/stdin:5 SC1090 Can't follow non-constant source. Use a directive to specify location.
```

### Expected behavior

Either drop the error entirely or provide a valid way to work around it.

### Version

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version`:**

```bash
$ docker run --rm hadolint/hadolint:latest hadolint --version
Haskell Dockerfile Linter v1.17.0-0-g56fd3e1
```

**Comment by ianfixes** [comment]
I checked whether (based on #76) it would be possible to solve this as follows:

```Dockerfile
FROM alpine:3
SHELL ["/bin/bash", "-c"]
ENV NVM_DIR=/usr/local/nvm
RUN \
  # shellcheck source=/dev/null
  source $NVM_DIR/nvm.sh
```
^ same error

```Dockerfile
FROM alpine:3
SHELL ["/bin/bash", "-c"]
ENV NVM_DIR=/usr/local/nvm
RUN \
  # shellcheck source=/dev/null \
  source $NVM_DIR/nvm.sh
```
^ same error, and VS code thinks it's a badly-formed Dockerfile

**Comment by zemanlx** [comment]
Thank you for reporting this issue. Including comments into script that will be generated from Dockerfile is tricky and workaround from shellcheck wiki won't work for hadolint, but you can achieve the same behaviour by ignoring SC1090 for that line 

```Dockerfile
FROM alpine:3
SHELL ["/bin/bash", "-c"]
ENV NVM_DIR=/usr/local/nvm
# hadolint ignore=SC1090
RUN source $NVM_DIR/nvm.sh
```

Does this solve your issue?

**Comment by ianfixes** [comment]
At the moment I am simply solving it via the command line with `hadolint --ignore SC1090`.  I expect to do this until the SC1090 error is removed from hadolint's checks or a useful workaround is presented.  (In other words, I dislike `# hadolint ignore=SC1090` since it does nothing for the quality or readability of the Dockerfile.)

**Comment by lorenzo** [comment]
@ianfixes would you like to submit a default rule ignore for this? It should be almost trivial and I can guide you in the process

**Comment by ianfixes** [comment]
I assume it's a one-line fix https://github.com/hadolint/hadolint/blob/master/src/Hadolint/Shell.hs#L82-L84

**Comment by lorenzo** [comment]
That’s exactly right

**Comment by ianfixes** [comment]
Submitted

**Comment by FrangaL** [comment]
ENV SHELLCHECK_OPTS='--shell=bash --exclude=SC1090'
